projects/aws/image-builder/cmd/build.go

```diff
@@ -155,12 +156,47 @@ func ValidateInputs(bo *builder.BuildOptions) error {
 					return err
 				}
 			}
+		case builder.AMI:
+			if err = json.Unmarshal(config, &bo.AMIConfig); err != nil {
+				return err
+			}
+			if bo.AMIConfig.CustomRole == "true" {
+				if (bo.AMIConfig.CustomRoleNameList == nil && bo.AMIConfig.CustomRoleNames == "") ||
+					(bo.AMIConfig.CustomRoleNameList != nil && bo.AMIConfig.CustomRoleNames != "") {
+					log.Fatalf("Exactly one of \"custom_role_name_list\" or \"custom_role_names\" must be provided")
+				}
+
+				if bo.AMIConfig.CustomRoleNameList != nil {
+					bo.AMIConfig.CustomRoleNames = strings.Join(bo.AMIConfig.CustomRoleNameList, " ")
```

Comment:
you probably want to nullify `CustomRoleNameList` here so it doesn't get marshaled back?